pkg/certs/cert-inspection/certgraphanalysis/collector.go

```diff
@@ -123,26 +125,27 @@ func gatherFilteredCerts(ctx context.Context, kubeClient kubernetes.Interface, a
 			}
 
 			details, err := InspectSecret(&secret)
-			if details != nil {
-				// TODO something like options.RewriteCertKeyPairDetails(details) (details, namespace, name)
-				//  this will allow us to rewrite node names and IP values
-				certs = append(certs, details)
-
-				inClusterResourceData.CertKeyPairs = append(inClusterResourceData.CertKeyPairs,
-					certgraphapi.PKIRegistryInClusterCertKeyPair{
-						SecretLocation: certgraphapi.InClusterSecretLocation{
-							Namespace: secret.Namespace,
-							Name:      secret.Name,
-						},
-						CertKeyInfo: certgraphapi.PKIRegistryCertKeyPairInfo{
-							OwningJiraComponent: secret.Annotations[annotations.OpenShiftComponent],
-							Description:         secret.Annotations[annotations.OpenShiftDescription],
-						},
-					})
-			}
 			if err != nil {
 				errs = append(errs, err)
+				continue
+			}
+			if details == nil {
+				continue
 			}
+			options.rewriteCertKeyPair(details)
+			certs = append(certs, details)
+
+			inClusterResourceData.CertKeyPairs = append(inClusterResourceData.CertKeyPairs,
+				certgraphapi.PKIRegistryInClusterCertKeyPair{
+					SecretLocation: certgraphapi.InClusterSecretLocation{
+						Namespace: secret.Namespace,
+						Name:      secret.Name,
+					},
+					CertKeyInfo: certgraphapi.PKIRegistryCertKeyPairInfo{
+						OwningJiraComponent: secret.Annotations[annotations.OpenShiftComponent],
+						Description:         secret.Annotations[annotations.OpenShiftDescription],
+					},
+				})
 		}
 	}
 
```

Comment:
the for loops below this are no longer needed, right?  I think it can be a followup, but should be done.